### ISSUE TITLE
Moving from SpatialConvolutionCUDA to SpatialConvolution possibly broke things?

### ISSUE THREAD

**Issue by chaddjohnson** [soumith/deepmind-atari#1]
Hi!

Thank you for your fork of this project, and thank you for all your brilliant work on Torch.

I cloned this repository and ran it on one of my Titan Z GPUs for two to three days straight. It's been learning to play Breakout. However, when I use the resulting network file, enable graphics, and use qlua to display visual output, the agent seems to play no more intelligently than an untrained version -- it never scores more than a few points in a row. Here are the steps I took ([source](http://superintelligence.ch/deepmind/)):
1. Cloned deepmind-atari.
2. Changed `../torch/bin/luajit train_agent.lua $args` to `../torch/bin/qlua train_agent.lua $args` in run_gpu.
3. Changed `display=false` to `display=true` in `torch/share/lua/5.1/alewrap/AleEnv.lua`
4. Ran `./run_gpu breakout` for two to three days. This produced the network file dqn/DQN3_0_1_breakout_FULL_Y.t7.
5. Changed `netfile="\"convnet_atari3\""` in run_gpu to `netfile="\"DQN3_0_1_breakout_FULL_Y.t7\""`.

My machine reached approximately 12 million steps. Maybe this is not enough?

I'm concerned that change https://github.com/soumith/deepmind-atari/commit/88fffeaa04724d1b34f2586f8fc08ebc44256cb0 may have broken this demo. Is this at all possible? Or do you have any suggestions on what might be wrong? I greatly appreciate any input!

**Comment by ghost** [comment]
@chaddjohnson  If you are interested in working on the deep-q-rl, may I suggest you join this Google group,

https://groups.google.com/forum/#!forum/deep-q-learning

I was a founding member, and the folks there have been working on it since xmas. I think you'll get much more support there - especially for the Python implementation?

**Comment by chaddjohnson** [comment]
I figured out how to make this work. Basically I reverted change 88fffea and then used forked versions of cunn, cutorch, torch7, and nn. I then reverted changes to these dependencies, removing changes after April 6, 2015, as change 88fffea occurred on April 7, 2015, and I modified install_dependencies.sh to install the dependencies from local directories. The simulation now learns as it should. I'm not sure if the issue was due to change 88fffea or a change in one of the dependencies I forked.

@AjayTalati I will take a look at that! Thank you for the heads up.

**Comment by chaddjohnson** [comment]
So, I ran run_gpu again with Breakout using the latest, unmodified code from this repository, and learning occurred with no problems. I'm not sure why it wasn't working before. I did upgrade the RAM from 16GB to 32GB in my machine, and this eliminated heavy usage of swap, so maybe that was a contributing factor.

Anyway, thank you for creating this fork! Time to reverse-engineer this software (and its dependencies) and make it even better.
